README.md

```diff
@@ -35,6 +35,45 @@ We have created a Cloud Shell with all of the tools and dependencies needed to r
 
 If you need support getting your Plural deployment up and running, join the [Plural Discord here!](https://discord.gg/pluralsh)
 
+## Architecture 
+
+The Plural architecture has three main components:
+
+* Plural API and Catalog site (available at [https://app.plural.sh](https://app.plural.sh))
+* Plural CLI and Git SCM to maintain the state of a user's applications
+* Plural console for management of all plural applications on your infrastructure
+
+At a high level, the interactions between all three components look something like this:
+
+![](</assets/reference/architecture.png>)
+
+### Plural API
+
+The primary responsibility of the Plural API is to store the packages needed for application installation - terraform, helm - and ingesting high-level dependency information about them. This allows us to properly sequence installations. It also serves as a publish-subscribe layer to communicate updates to clusters that have installed those applications, and can leverage the dependency information ingested to delay updates until a cluster has caught up with all the necessary dependencies.
```

Comment:
This is heavy to read. IMO could be simplified.

Reply:
agreed, this is a bit out of scope, so I'll do this in another PR.